crates/compiler/load_internal/src/file.rs

```diff
@@ -3838,6 +3838,36 @@ struct HeaderOutput<'a> {
     opt_platform_shorthand: Option<&'a str>,
 }
 
+fn ensure_roc_file<'a>(filename: &PathBuf, src_bytes: &[u8]) -> Result<(), LoadingProblem<'a>> {
+    match filename.extension() {
+        Some(ext) => {
+            let roc_ext = std::ffi::OsStr::new("roc");
+            if roc_ext != ext {
+                return Err(LoadingProblem::FileProblem {
+                    filename: filename.clone(),
+                    error: io::ErrorKind::Unsupported,
+                });
+            }
+        }
+        None => {
+            let index = src_bytes
+                .iter()
+                .position(|a| *a == b'\n')
+                .unwrap_or(src_bytes.len());
```

Comment:
Is there a better way of checking the first line for a shebang? This works, but I feel like we should be using the parser for this, not manually comparing strings

Reply:
Based on a quick search I think the parser parses does not record any info about a possible shebang. It treats it like a regular comment. I understand your desire but I think this is fine.

Reply:
Oh, right. I seem to remember one of the reasons for making comments start with `#` was for exactly this case